Explicit print options while dumping HLO

shraiysh opened the pull request:
We are changing the `HloModule::ToString` function to consider the debug options related to printing like, `xla_dump_hlo_as_long_text`, `xla_dump_large_constants` etc. These options are already embedded into the `HloModule` and are set via the command line and so, the `ToString` function should default to these functions. 

We are also changing the default value of HloPrintOptions: `set_print_operand_shape(false)` and `set_print_large_constants(true)`.

The related changes in XLA are
https://github.com/openxla/xla/pull/22614 and https://github.com/openxla/xla/pull/22800.

This breaks some of the tests in tensorflow because it relies on the `ToString` function and the default print options. As a temporary fix, we are making the default print options explicit here. Once the XLA change gets merged, we can remove this and fix the tests (or not, depending on what the TF community decides, but the behavior of the tool will be intentional.)

allanrenucci commented on tensorflow/compiler/mlir/tensorflow/utils/tf_xla_mlir_translate.cc:
```diff
@@ -114,7 +114,7 @@ mlir::LogicalResult PrintHloModuleText(
   xla::HloModule* hlo_module = status_or_hlo_module.value().get();
 
-  output << hlo_module->ToString();
+  output << hlo_module->ToString(xla::HloPrintOptions());
```
Nit, this avoid constructing a temporary value:
```suggestion
  output << hlo_module->ToString(xla::HloPrintOptions::Default());
```

shraiysh replied:
I have edited it to set two of the print options. Right now, those values are the default values, of those print options, but they are going to change after https://github.com/openxla/xla/pull/22800. Is this okay?

shraiysh committed:
Explicit print options while dumping HLO

We are changing the `HloModule::ToString` function to consider the debug options related to printing like, `xla_dump_hlo_as_long_text`, `xla_dump_large_constants` etc. These options are already embedded into the `HloModule` and are set via the command line and so, the `ToString` function should default to these functions.

We are also changing the default value of HloPrintOptions: `set_print_operand_shape(false)` and `set_print_large_constants(true)`.

The related changes in XLA are
https://github.com/openxla/xla/pull/22614 and https://github.com/openxla/xla/pull/22800.

This breaks some of the tests in tensorflow because it relies on the `ToString` function and the default print options. As a temporary fix, we are making the default print options explicit here. Once the XLA change gets merged, we can remove this and fix the tests (or not, depending on what the TF community decides, but the behavior of the tool will be intentional.)

akuegel reviewed (approved):
Thank you :)
